mod-pubsub-server/src/main/java/org/folio/services/impl/KafkaTopicServiceImpl.java

```diff
@@ -30,18 +30,18 @@ public KafkaTopicServiceImpl(@Autowired KafkaAdminClient kafkaAdminClient, @Auto
   }
 
   @Override
-  public Future<Boolean> createTopics(List<String> eventTypes, String tenantId) {
-    Promise<Boolean> promise = Promise.promise();
+  public Future<Void> createTopics(List<String> eventTypes, String tenantId) {
+    Promise<Void> promise = Promise.promise();
     List<NewTopic> topics = eventTypes.stream()
       .map(eventType -> new NewTopic(new PubSubConfig(kafkaConfig.getEnvId(), tenantId, eventType).getTopicName(), kafkaConfig.getNumberOfPartitions(), (short) kafkaConfig.getReplicationFactor()))
       .collect(Collectors.toList());
     kafkaAdminClient.createTopics(topics, ar -> {
       if (ar.succeeded()) {
         LOGGER.info("Created topics: [{}]", StringUtils.join(eventTypes, ","));
-        promise.complete(true);
+        promise.complete();
       } else {
         LOGGER.info("Some of the topics [{}] were not created. Cause: {}", StringUtils.join(eventTypes, ","), ar.cause().getMessage());
-        promise.complete(false);
+        promise.fail(ar.cause());
```

Comment:
It is correct to return a failure if some of the topics were not created.

However, the old code returned false and false had been ignored. Now some other tests fail because trying to create a topic twice fails. The tests need to be adjusted: Either be sure not to re-create a topic, or ignore the failure in the other test.

Reply:
you're right, right now there are failing 12 tests out of 85 because of the logic here.
There are two options to handle the issue:
1) Using cash to check already created topics
2) Adjust failed tests
@julianladisch are you OK if I change failed tests?